src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -67,1008 +50,333 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
-	                   addr.Latest_End_Date__c = null;
-	               }
-	            }
-	            
-	            // BEFORE UPDATE
-	            if (triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
-	                if (addr.Default_Address__c != addrOld.Default_Address__c) {
-	                   // when an address changes Default, we update its latest date fields
-	                    if (addr.Default_Address__c) {
-	                       addr.Latest_Start_Date__c = System.today();
-	                       addr.Latest_End_Date__c = null;
-	                    } else {
-	                       addr.Latest_End_Date__c = System.today();                        
-	                    }
-	                }
-	            }
+        if (!hasRunAddrTrigger) {
+	        // first go through all new/modified Addresses, and collect the list of parents to consider.
+	        // map of parents, and their default Address 
+	        Map<Id, Address__c> mapParentIdAddr = new Map<Id, Address__c>(); 
 	
-	            // AFTER INSERT
-	            if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
-	                // a new address that is marked default or seasonal needs to propogate to the household
-	                if (addr.Default_Address__c || isSeasonalAddr(addr)) 
-	                    mapAccIdAddr.put(addr.Parent_Account__c, addr);
-	            }
-	            
-	            // AFTER UPDATE
-	            if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
-	                // an address that is marked default needs to propogate to the household
-	                if ((addr.Default_Address__c && (addr.Default_Address__c != addrOld.Default_Address__c)) 
-	                    // any seasonal address also need to be considered
-	                    || isSeasonalChanged(addr, addrOld)) {
-	                    mapAccIdAddr.put(addr.Parent_Account__c, addr);
-	                } else if (isAddressChanged(addr, addrOld, true)) {
-	                    if (addr.Default_Address__c)
-	                        mapAccIdAddr.put(addr.Parent_Account__c, addr);
-	                    else    // address changed, but not default, still need to update any contacts using it as an override
-	                        mapAddrIdAddr.put(addr.Id, addr);      
-	                } 
+	        // AFTER DELETE
+	        if (triggerAction == TDTM_Runnable.Action.AfterDelete) {
+	            for (SObject so : listOld) {
+	                ID parentAccId = ((Address__c)so).Parent_Account__c;
+	                ID parentConId = ((Address__c)so).Parent_Contact__c;
+	                if (parentAccId != null) {
+	                    mapParentIdAddr.put(parentAccId, null);
+	                } else if(parentConId != null) {
+	                    mapParentIdAddr.put(parentConId, null);
+	                }
 	            }
-	            
-	            // moving to the next item
-	            i++;            
+	        } else {
+		        integer i = 0;        
```

Comment:
This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.